components/configurations-generator/before-commit.sh

```diff
@@ -0,0 +1,93 @@
+#!/usr/bin/env bash
+
+readonly CI_FLAG=ci
+
+RED='\033[0;31m'
+GREEN='\033[0;32m'
+INVERTED='\033[7m'
+NC='\033[0m' # No Color
+
+echo -e "${INVERTED}"
+echo "USER: " + $USER
+echo "PATH: " + $PATH
+echo "GOPATH:" + $GOPATH
+echo -e "${NC}"
+
+##
+# DEP ENSURE
+##
+dep ensure -v --vendor-only
+ensureResult=$?
+if [ ${ensureResult} != 0 ]; then
+	echo -e "${RED}✗ dep ensure -v --vendor-only${NC}\n$ensureResult${NC}"
+	exit 1
+else echo -e "${GREEN}√ dep ensure -v --vendor-only${NC}"
+fi
+
+##
+# GO BUILD
+##
+buildEnv=""
+if [ "$1" == "$CI_FLAG" ]; then
+	# build binary statically
+	buildEnv="env CGO_ENABLED=0"
+fi
+
+${buildEnv} go build -o generator ./cmd/generator/
+
+goBuildResult=$?
+if [ ${goBuildResult} != 0 ]; then
+	echo -e "${RED}✗ go build${NC}\n$goBuildResult${NC}"
+	exit 1
+else echo -e "${GREEN}√ go build${NC}"
+fi
+
+##
+# DEP STATUS
+##
+echo "? dep status"
+depResult=$(dep status -v)
+if [ $? != 0 ]; then
+	echo -e "${RED}✗ dep status\n$depResult${NC}"
+	exit 1
+else echo -e "${GREEN}√ dep status${NC}"
+fi
+
+##
+# GO TEST
+##
+echo "? go test"
+go test ./...
+# Check if tests passed
+if [ $? != 0 ]; then
+	echo -e "${RED}✗ go test\n${NC}"
+	exit 1
+else echo -e "${GREEN}√ go test${NC}"
+fi
+
+goFilesToCheck=$(find . -type f -name "*.go" | egrep -v "\/vendor\/|_*/automock/|_*/testdata/|/pkg\/|_*export_test.go")
```

Comment:
why do you skip go files in the `/pkg` directory? I would say that we should format them, as in most cases they contain code written by us.

Reply:
Good call, fixed.